logger/src/metrics.rs

```diff
@@ -116,6 +116,40 @@ impl Serialize for SharedMetric {
     }
 }
 
+/// Representation of a metric that is expected to be incremented from more than one thread, and
+/// which can be only 0 or 1 (i.e. is binary).
+// The synchronization is identical to SharedMetric.
+#[derive(Default)]
+pub struct BinarySharedMetric(AtomicBool, AtomicBool);
+
+impl Metric for BinarySharedMetric {
+    fn add(&self, value: usize) {
+        if value > 0 {
+            self.0.store(true, Ordering::Relaxed);
+        }
+    }
+
+    fn count(&self) -> usize {
+        // The Rust Standard specifies that converting a boolean to an integer result in 
+        // either 0 or 1, as we want for this application.
+        self.0.load(Ordering::Relaxed) as usize
+    }
+}
+
+// This implementation is identical to the implementation of Serialize
+// for SharedMetric.
+impl Serialize for BinarySharedMetric {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let snapshot = self.0.load(Ordering::Relaxed);
+        let res = serializer.serialize_u64(snapshot as u64 - self.1.load(Ordering::Relaxed) as u64);
```

Comment:
I don't think computing a delta here (`snapshot as u64 - self.1.load(Ordering::Relaxed) as u64`) is correct.

We can always just use `self.0` and totally ignore `self.1` since we need absolute values not deltas.

Reply:
Actually we should also use better naming here.

We might want binary metrics that are persistent throughout the whole lifetime of the process (printing a `true` value will not reset it to `false`), and we might want binary metrics that should have the scope of a single metrics-report-period (like the other numeric metrics have, which get reset after each report).

These are differently behaving metrics and their particular flavor should be present in the name.

Reply:
Ok, should this idea be applied to the SharedMetrics as well ?

Reply:
I don't think reworking the other `SharedMetrics` is in the scope of this PR, as they're all reset-on-flush.

Reply:
Hi, just wondering, what's the use case for the changes introduced in this PR (besides removing `SimpleMetric` because it isn't used)? I've had a quick look at  #1173, but it's still not very clear why these additions are required :-s

Reply:
In order to test the SIGBUS/SIGSEGV handler from #1173, it is necessary to introduce metrics whose value can be either 0 or 1. I was thinking that [this](https://github.com/firecracker-microvm/firecracker/pull/1173#discussion_r302535485) discussion implies that using general purpose metrics for such values is not best practice, since it makes the code less clear. Basically encoding the boolean-ness of a metric in the type will make it more clear what those metrics are used for.